packages/manager/apps/account-disable-2fa/src/utils/invisible-challenge.ts

```diff
@@ -0,0 +1,165 @@
+import { AxiosInstance, InternalAxiosRequestConfig } from 'axios';
+import { IframeLoaderPayload, loadIframe, removeIframe } from '@/utils/iframe';
+
+export type ChallengeApiInterceptorChallengeResponse = {
+  challenge_response: string;
+};
+
+type Deferred<T> = {
+  promise: Promise<T>;
+  resolve?: (value: T) => void;
+  reject?: (reason?: ChallengeFailureReasons) => void;
+};
+
+export enum ChallengeFailureReasons {
+  NoIframe,
+  Timeout,
+}
+
+export const CHALLENGE_IFRAME_ID = 'ovh_security_challenge';
+export const TTL_TO_WAIT_IFRAME_RESPONSE = 1000;
+export const INVISIBLE_CHALLENGE_ERROR_CLASS =
+  'Client::BadRequest::ChallengeRequired';
+export const IFRAME_STYLE = 'position: absolute; width:0; height:0; border:0;';
+
+export class ChallengeApiInterceptor {
+  private apiClient: AxiosInstance;
+
+  private iframe: HTMLIFrameElement | null;
+
+  private deferWaitingChallenge: Deferred<unknown> | null;
+
+  private abortTimeout: number;
+
+  constructor(apiClient: AxiosInstance) {
+    this.apiClient = apiClient;
+    this.iframe = null;
+    this.deferWaitingChallenge = null;
+    this.abortTimeout = null;
+  }
+
+  public attachMessageListener() {
+    window.addEventListener(
+      'message',
+      (message: MessageEvent) => this.onIframeMessage(message),
+      false,
```

Comment:
**nitpick:** Anyways defaults to `false` (ref: https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#usecapture) . So better to remove.
```suggestion
```